### PR TITLE
fix: persist data on android

### PR DESCRIPTION
**Pull request by tommasini**
## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/redux/storage/MigratedStorage.ts [hunk 1]
```diff
@@ -10,6 +10,20 @@ const createMigratedStorage = (reducerName: string) => ({
         // Using the new storage system
         return res;
       }
```
Comment (MarioAslau) [inline app/redux/storage/MigratedStorage.ts:12]: Could we add a TODO comment to not forget to remove this part of the code once all users have moved to the new version ?

**Review by MarioAslau — DISMISSED** [review]
Screenshot 2024-02-08 at 12 41 59

File: app/redux/storage/MigratedStorage.ts (hunk 1 above)
Comment (tommasini) [reply app/redux/storage/MigratedStorage.ts:12]: Yess!!

File: app/store/persistConfig.ts [hunk 2]
```diff
@@ -55,6 +58,8 @@ export const rootPersistConfig = createPersistConfig({
   timeout: TIMEOUT,
   writeFailHandler: (error: Error) =>
     Logger.error(error, { message: 'Error persisting data' }), // Log error if saving state fails
+  customStateReconcilier: (inboundState: any, originalState: any) =>
```
Comment (Cal-L) [inline app/store/persistConfig.ts:61]: I think it's safer to fallback to this reconciler by default since it handles merging levels dynamically regardless of size. We can ensure that future changes won't run into the same issue that we have. Nonetheless, it's always good practice to reduce the number of levels of nested data here.

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1373,11 +1373,12 @@
 					"@executable_path/Frameworks",
 				);
 				LIBRARY_SEARCH_PATHS = (
+					"$(SDKROOT)/usr/lib/swift",
```
Comment (Cal-L) [inline ios/MetaMask.xcodeproj/project.pbxproj:1376]: Remove unrelated changes as well as version bumps

**Review by Cal-L — COMMENTED** [review]
left some comments

**Review by NicolasMassart — COMMENTED** [review]
Simple and clean fix.
Same as @Cal-L I would prefer version and unrelated stuff out of this PR.
Also, could you fill the PR description and explain how the merge call fixes the issue? Why is there a fallback? A merge should never fail, right?
